### DISCUSSION TITLE
UART and Interrupts will be available in MicroPython port for PSoC6?

### DISCUSSION THREAD

**Discussion by Antonia-place** [Infineon/micropython#74] [General]
Hi,

I want to know if UART and Interrupts (GPIO or others) will be available in MicroPython for PSoC6.
Until now I see only I2C, SPI, GPIO (but nothing about interrupts) and ADC is available in MicroPython v0.4.0 for PSoC6.
https://ifx-micropython.readthedocs.io/en/latest/psoc6/quickref.html#

Thank you for support,
Mihai

**Comment by jaenrig-ifx** [comment]
Hi @Antonia-place,

We are starting now to work on **interrupts** (which affects several machine classes Pin, Timer, ...) and **I2S** enablement. UART is already scheduled after.
Concrete release dates I cannot promise :) 

Regards,
Juan

**Reply by Antonia-place** [reply]
ok, thank you and good luck! 
You are doing a good job with MicroPython on PSoC6!!!

**Reply by jaenrig-ifx** [reply]
A few things ahead to enable all the potential of the PSoC6 🚀 
Thanks a lot for contributing with your feedback and support! 👯‍♀️

**Comment by jaenrig-ifx** [comment]
UART is now enabled:
https://github.com/Infineon/micropython/releases/tag/v0.11.0
